server/src/test/java/org/elasticsearch/search/fetch/subphase/FieldFetcherTests.java

```diff
@@ -1162,7 +1162,7 @@ public void testFetchMetadataFieldWithSourceDisabled() throws IOException {
             IndexSearcher searcher = newSearcher(iw);
             LeafReaderContext readerContext = searcher.getIndexReader().leaves().get(0);
             fieldFetcher.setNextReader(readerContext);
-            SourceLookup sourceLookup = new SourceLookup();
+            SourceLookup sourceLookup = new SourceLookup.Static();
             sourceLookup.setSegmentAndDocument(readerContext, 0);
```

Comment:
Oh this is fun. `Static` isn't going to be able to do anything with the `readerContext` or the docId. But this is how we pass those things into the fetching framework, right?